content/guides/typescript-for-solid.mdx

```diff
@@ -0,0 +1,707 @@
+
+# TypeScript for Solid
+
+## Overview
+
+[TypeScript](https://www.typescriptlang.org/) is a superset of JavaScript that enhances code reliability and predictability through the introduction of [static types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html). 
+While JavaScript code can be directly used in TypeScript, the added type annotations in TypeScript provide clearer code structure and documentation, making it more accessible for developers.
+
+By leveraging standard JSX, Solid facilitates seamless TypeScript interpretation. 
+Moreover, with Solid's built-in types for the API heighten accuracy. 
+
+For developers eager to get started, we offer TypeScript templates our [GitHub](https://github.com/solidjs/templates/). 
+
+## Configuring TypeScript
+
+When integrating TypeScript with the Solid JSX compiler, there are crucial settings to make for a seamless interaction:
+
+1. `"jsx": "preserve"` in the `tsconfig.json` retains the original JSX form.
+2. `"jsxImportSource": "solid-js"` designates SolidJS as the source of JSX types.
+
+For a basic setup, your `tsconfig.json` should resemble:
+
+```json
+{
+  "compilerOptions": {
+    "jsx": "preserve",
+    "jsxImportSource": "solid-js"
+  }
+}
+```
+
+For projects with diverse JSX sources, such as a blend of React and Solid, some flexibility does exist. 
+While it's possible to set a default `jsxImportSource` in `tsconfig.json` to corresponds with the majority of your files, TypeScript also allows file-level overrides. 
+Utilizing specific pragmas within `.tsx` files facilitates this:
+
+```javascript
+/** @jsxImportSource solid-js */
+```
+or, if using React:
+
+```javascript
+/** @jsxImportSource react */
+```
+
+Opting for the React JSX pragma means having React and its associated dependencies fully integrated into the project.
+Furthermore, it makes sure the project's architecture is primed for React JSX file handling is vital.
+
+## Migrating from JavaScript to TypeScript
+
+Transitioning from JavaScript to TypeScript in a Solid project offers the benefits of static typing. Here are some of the key steps and considerations you should bear in mind during the migration:
+
+1. Install TypeScript into your project.
+
+```bash
+npm install --save-dev typescript
+```
+
+2. Run `npx tsc --init` to generate a `tsconfig.json` file.
+
+3. Update the contents of your `tsconfig.json` to match Solid's configuration:
+
+```json
+{
+  "compilerOptions": {
+    "strict": true,
+    "target": "ESNext",
+    "module": "ESNext",
+    "moduleResolution": "node",
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "jsx": "preserve",
+    "jsxImportSource": "solid-js",
+    "types": ["vite/client"],
+    "noEmit": true,
+    "isolatedModules": true
+  }
+}
+```
+
+4. Create a TypeScript or `.tsx` file to test the setup.
+
+```typescript
+import { Component } from "solid-js";
+
+function MyTsComponent(): Component {
```

Comment:
```suggestion
function MyTsComponent(): JSX.Element {
```
The `Component` type refers to the function rather than the return type. Alternatively, 
```suggestion
const MyTsComponent: Component = () => {
```
You could also use `VoidComponent` here if you'd like. Personally I prefer not to use it but it's a matter of style.